src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/VersionPropsPlugin.java

```diff
@@ -96,92 +99,65 @@ public void configure(Map<String, String> config) {
 
-                        };
-                    else
-                        return super.visitMethod(access, name, desc, sig, xs);
+                                    @Override
+                                    public void accept(CodeBuilder cob, CodeElement coe) {
+                                        if (coe instanceof Instruction ins) switch (ins.opcode()) {
+                                            case LDC, LDC_W, LDC2_W -> {
+                                                flushPendingLDC(cob);
+                                                pendingLDC = coe;
+                                            }
+                                            case INVOKEVIRTUAL, INVOKESPECIAL, INVOKESTATIC, INVOKEINTERFACE -> {
+                                                flushPendingLDC(cob);
+                                                cob.accept(coe);
+                                            }
+                                            case GETSTATIC, GETFIELD, PUTFIELD -> {
+                                                flushPendingLDC(cob);
+                                                cob.accept(coe);
+                                            }
+                                            case PUTSTATIC -> {
+                                                if (((FieldInstruction)coe).name().equalsString(field)) {
+                                                    // assert that there is a pending ldc
+                                                    // for the old value
+                                                    if (pendingLDC == null) {
+                                                        throw new AssertionError("No load " +
+                                                            "instruction found for field " + field +
+                                                            " in static initializer of " +
+                                                            VERSION_PROPS_CLASS);
+                                                    }
+                                                    // forget about it
+                                                    pendingLDC = null;
+                                                    // and add an ldc for the new value
+                                                    cob.constantInstruction(value);
+                                                    redefined = true;
+                                                } else {
+                                                    flushPendingLDC(cob);
+                                                }
+                                                cob.accept(coe);
+                                            }
+                                            default -> cob.accept(coe);
+                                        }
+                                    }
+                                });
+                            }
+                        });
```

Comment:
```suggestion
                        });
                } else {
                    clb.with(cle);
```

Other `ClassElement`s need to be added to the builder; otherwise they are dropped.  

One more thing I notice is that the attributes for example `LineNumberTable` attribute in `<clinit>` method after transformed are dropped.   But this plugin should only transform the code and leave everything else touched.

Reply:
Yes, this transformation is incomplete and not using the latest Classfile API patterns.
I'll update it, thanks.